core-v1/src/test/scala/SimpleJustifiedSelectSpec.scala

```diff
@@ -0,0 +1,162 @@
+      _.get(_.select(_.value))
+    }
+    val expectedWrapped = Some(
+      Justified.bySelection(
+        expected.value,
+        DataPath.empty.atField("opt").atIndex(0).atField("value"),
+        Justified.byConst(fixture),
+      ),
+    )
+    val obtained = expr.run()
+    assertEquals(obtained, expectedWrapped)
+  }
+
+  test("Cannot use the Lens.select operation to map over elements of an Option") {
+    val fixture = NestedSelectable("optDefined", opt = Some(empty))
+    val _ =
+      fixture.const.get(_.select(_.opt)).map(_.get(_.select(_.value))) // correct
+    val message = compileErrors {
+      "fixture.const.get(_.select(_.opt.map(_.value)))" // incorrect
+    }
+    assert(message contains VariantLensMacros.InvalidDataPathMessage)
+  }
+
+  test("Select a nested Seq value of a const") {
+    val firstValue = NestedSelectable("firstValue")
+    val secondValue = NestedSelectable("secondValue")
+    val fixture = NestedSelectable("seqDefined", seq = Seq(firstValue, secondValue))
+    val expr = fixture.const.get(_.select(_.seq)).map(_.get(_.select(_.value)))
+    val observed = expr.run()
+    val expected = Seq(
+      Justified.bySelection(
+        firstValue.value,
+        DataPath.empty.atField("seq").atIndex(0).atField("value"),
+        Justified.byConst(fixture),
+      ),
+      Justified.bySelection(
+        secondValue.value,
+        DataPath.empty.atField("seq").atIndex(1).atField("value"),
+        Justified.byConst(fixture),
+      ),
+    )
+    assertEquals(observed, expected)
+  }
+
+  test("Cannot use the Lens.select operation to map over elements of a Seq") {
+    val fixture = NestedSelectable("seqDefined", seq = Seq(empty, empty))
+    val _ =
+      fixture.const.get(_.select(_.seq)).map(_.get(_.select(_.value))) // correct
+    val message = compileErrors {
+      "fixture.const.get(_.select(_.seq.map(_.value)))" // incorrect
+    }
+    assert(message contains VariantLensMacros.InvalidDataPathMessage)
+  }
+
+  test("Select the values of a nested Map value of a const") {
+    val first = NestedSelectable("firstValue")
+    val second = NestedSelectable("secondValue")
+    val fixture = NestedSelectable("mapDefined", map = Map("one" -> first, "two" -> second))
+    val expr = fixture.const.get(_.select(_.map).to[Seq]).map { kv =>
+      kv.get(_.at(Nat._1).select(_.value))
+    }
+    val observed = expr.run()
+    val expectedWrapped = Seq(
+      Justified.bySelection(
+        first.value,
+        // TODO: How to keep the original map keys / remove the tuple keys here?
+        DataPath.empty.atField("map").atIndex(0).atKey("1").atField("value"),
+        Justified.byConst(fixture),
+      ),
+      Justified.bySelection(
+        second.value,
+        // TODO: How to keep the original map keys / remove the tuple keys here?
+        DataPath.empty.atField("map").atIndex(1).atKey("1").atField("value"),
+        Justified.byConst(fixture),
+      ),
```

Comment:
I'm not gonna worry about these TODOs now. I'll investigate this later when I care more about justification. I think I need to rework some things about the lens / WrapSelect implicits, but it isn't likely to change the interface, so I think this is fine to leave here for now.